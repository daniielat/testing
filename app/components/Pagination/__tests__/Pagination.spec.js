const React = require('react');
const Pagination = require('..');
const { render, screen, fireEvent } = require('@testing-library/react');

describe('El componente Pagination', () => {
    let component;
    const i18n = { gettext: text => text };
    const setOffset = jest.fn((offset) => offset = offset + 10)
    
    beforeEach(() => {
        component = render(<Pagination i18n={i18n} offset={0} setOffset={setOffset}/>);
    });

    it('Renderiza', () => {
        const { asFragment } = component;
        expect(asFragment()).toMatchSnapshot();
    });

    // it('Aumenta el valor de offser al clickear el botÃ³n `Siguiente`', () => {
    //     const button = screen.getByRole('button');
    //     fireEvent.click(button);

    //     expect(offset).toEqual(10);
    // });
});