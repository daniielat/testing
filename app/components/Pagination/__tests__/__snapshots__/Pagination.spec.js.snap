// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`El componente Pagination Renderiza 1`] = `
<DocumentFragment>
  <section>
    <button
      class="andes-button andes-button--large andes-button--quiet"
      type="button"
    >
      <span
        class="andes-button__content"
      >
        Siguiente
      </span>
    </button>
  </section>
</DocumentFragment>
`;
